app/screens/channel_files/channel_files.tsx

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {FlatList, ListRenderItemInfo, StyleSheet, View} from 'react-native';
+import {Edge, SafeAreaView, useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import {searchFiles} from '@actions/remote/search';
+import Loading from '@components/loading';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import {useImageAttachments} from '@hooks/files';
+import {showMobileOptionsBottomSheet} from '@screens/home/search/results/file_options/mobile_options';
+import Toasts from '@screens/home/search/results/file_options/toasts';
+import FileResult from '@screens/home/search/results/file_result';
```

Comment:
I think these should be moved from `@screens/home/search`  to some utility file an components in common

Reply:
What should be the ideal place to keep common code.